internal/cmd/server.go

```diff
@@ -0,0 +1,149 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/datadog/orchestrion/internal/filelock"
+	"github.com/datadog/orchestrion/internal/jobserver"
+	"github.com/datadog/orchestrion/internal/jobserver/client"
+	"github.com/datadog/orchestrion/internal/log"
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/urfave/cli/v2"
+)
+
+var Server = &cli.Command{
+	Name:        "server",
+	Usage:       "Start an Objectsrion job server.",
+	Description: "The job server is used to remove duplicated processing that can occur when isntrumenting large applications, due to how Orchestrion injects new dependencies that the go toolchain was initially not aware of.\n\nUsers do not normally need to use this command directly, as Orchestrion automatically manages servers during runtime.",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "url-file",
+			Usage: "Write a file containing the ClientURL for this server once it is ready to accept connections",
+		},
+		&cli.IntFlag{
+			Name:        "port",
+			Usage:       "Choose a port to listen on",
+			Value:       -1,
+			DefaultText: "random",
+		},
+		&cli.DurationFlag{
+			Name:  "inactivity-timeout",
+			Usage: "Automatically shut down after a period without any connected client",
+			Value: time.Minute,
+		},
+		&cli.BoolFlag{
+			Name:  "nats-logging",
+			Usage: "Enable NATS server logging",
+		},
+	},
+	Hidden: true,
+	Action: func(c *cli.Context) error {
```

Comment:
There is about 70% of this function being behind `urlFile != ""` two if statements. How about reversing the logic here or at least moving it in another function dedicated to the non-happy path ?